core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -113,6 +116,21 @@ public JoinDefinition define(AutoJoinerContext context) {
       } else {
         joinBuilder.setOutputSchemaName("join.output");
       }
+
+      if (conf.isDistributionEnabled()) {
+        if (!Strings.isNullOrEmpty(conf.getDistributionStageName()) && conf.getDistributionFactor() != null) {
+          joinBuilder.setDistributionFactor(conf.getDistributionFactor(), conf.getDistributionStageName());
+        } else {
+          if (conf.getDistributionFactor() == null) {
+            collector.addFailure("Distribution Size is a required value if distribution is enabled.", "")
+              .withConfigProperty("distributionFactor");
+          }
+          if (Strings.isNullOrEmpty(conf.getDistributionStageName())) {
+            collector.addFailure("Skewed Stage name is a required value if distribution is enabled.", "")
+              .withConfigProperty("distributionStageName");
+          }
+        }
+      }
       return joinBuilder.build();
     } catch (InvalidJoinException e) {
```

Comment:
if there is some error with the distribution values (ex: not an inner or left outer join), they should be handled here, with the proper property highlighted. This will require some more specific error in your other CDAP PR.

Reply:
Noted

Reply:
Added new error typer and handling for errors